libafl/src/corpus/inmemory_ondisk.rs

```diff
@@ -312,29 +329,48 @@ where
         } else {
             // New testcase, we need to save it.
             let mut file_name = file_name_orig.clone();
-
             let mut ctr = 2;
-            let (file_name, lockfile_path) = loop {
-                let lockfile_name = format!(".{file_name}.lafl_lock");
-                let lockfile_path = self.dir_path.join(lockfile_name);
-
-                if OpenOptions::new()
-                    .write(true)
-                    .create_new(true)
-                    .open(&lockfile_path)
-                    .is_ok()
-                {
-                    break (file_name, lockfile_path);
-                }
-
-                file_name = format!("{file_name_orig}-{ctr}");
-                ctr += 1;
-            };
-
-            *testcase.file_path_mut() = Some(self.dir_path.join(&file_name));
-            *testcase.filename_mut() = Some(file_name);
-
-            fs::remove_file(lockfile_path)?;
+            // When the lock_file=true,Try to create lock file for new testcases
+            if self.lafl_lock {
+                let (new_file_name, lockfile_path) = loop {
+                    let lockfile_name = format!(".{file_name}.lafl_lock");
+                    let lockfile_path = self.dir_path.join(lockfile_name);
+
+                    if OpenOptions::new()
+                        .write(true)
+                        .create_new(true)
+                        .open(&lockfile_path)
+                        .is_ok()
+                    {
+                        break (file_name, lockfile_path);
+                    }
+
+                    file_name = format!("{file_name_orig}-{ctr}");
+                    ctr += 1;
+                };
+                fs::remove_file(lockfile_path)?;
+                file_name = new_file_name;
+                *testcase.file_path_mut() = Some(self.dir_path.join(&file_name));
+                *testcase.filename_mut() = Some(file_name);
+            } else {
+                // When the lock_file=false,Try to create file for new testcases
+                let new_file_name = loop {
```

Comment:
Ah actually if you still create a file (and just not name it .lafl_lock(), you will not be racey, the only issue is that we may produce empty files if we quit. I initially thought you would not check if the file exists, at all..
Interesting.
What's your use case then?

Reply:
I don't quite understand why an empty file is generated. I have experimented with the code locally and it will not affect the file generation.
```
use std::fs::OpenOptions;
use std::{thread, env};



fn create(){
    let mut ctr = 1;
    let mut file_name="file".to_owned();
    let new_file_name = loop {
        let file_path=env::current_dir().unwrap().join(&file_name);
        println!("111Trying with file name: {:?}", file_path);
        if OpenOptions::new()
            .write(true)
            .create_new(true)
            .open(&file_path)
            .is_ok()
        {
            println!("File created: {:?}", file_path);
            break file_name;
        }
        println!("File already exists: {:?}", file_path);
        let new_name=format!("{}-{}",file_name,ctr);
        file_name = new_name;
        println!("Trying with new file name: {:?}", file_name);
        ctr += 1;
    };
}




fn main() {
    let mut handles = vec![];
    
    for i in 0..5 {
        handles.push(thread::spawn(move || {
            create();
        }));
    }
    for handle in handles {
        handle.join().unwrap();
    }
}
```